packages/types/types/src/vector.ts

```diff
@@ -32,9 +32,13 @@ export interface IVectorShapeReader {
     readonly vectorProducer?: IVectorShapeProducer;
 }
 
-/** Capability to insert, replace, and remove items in a vector. */
+/**
+ * Capability to adjust the vector's length by inserting and/or removing items.  Use
+ * 'IVectorWriter' to assign values to the inserted items.  (The initial value of newly
+ * inserted items is implementation specific.)
+ */
 export interface IVectorShapeWriter {
-    splice(start: number, deleteCount: number, insertCount: number): void;
+    resize(start: number, deleteCount: number, insertCount: number): void;
```

Comment:
'resize' isn't a terrible name, but if there's something better that pops out at you let me know.

Reply:
Nothings pops outs. reshape?

Reply:
Yeah...  it's hard to beat 'splice' given that it matches the semantics of splice.  :-/

At least resize is "discoverable" in that once people give up looking for splice and resign themselves to resizing the vector and moving items themselves they'll find that resize can insert at any index.

Thanks for taking a look.  :-)